.circleci/merge_base_branch.sh

```diff
@@ -0,0 +1,28 @@
+#!/usr/bin/env bash
+
+set -e
+
+# CircleCI doesn't give CIRCLE_PR_NUMBER in the environment for non-forked PRs. Wonderful.
+# In this case, we just need to do some ~shell magic~ to strip it out of the PULL_REQUEST URL.
+echo 'export CIRCLE_PR_NUMBER="${CIRCLE_PR_NUMBER:-${CIRCLE_PULL_REQUEST##*/}}"' >> $BASH_ENV
+source $BASH_ENV
+
+if [[ -n "${CIRCLE_PR_NUMBER}" ]]
+then
+    # Update PR refs for testing.
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/head:pr/${CIRCLE_PR_NUMBER}/head"
+    FETCH_REFS="${FETCH_REFS} +refs/pull/${CIRCLE_PR_NUMBER}/merge:pr/${CIRCLE_PR_NUMBER}/merge"
+
+    # Retrieve the refs
+    git fetch -u origin ${FETCH_REFS}
+
+    # Checkout PR merge ref.
+    git checkout -qf "pr/${CIRCLE_PR_NUMBER}/merge"
+
+    # Test for merge conflicts.
+    git branch --merged | grep "pr/${CIRCLE_PR_NUMBER}/head" > /dev/null
+
+fi
```

Comment:
If we can't find a CIRCLE_PR_NUMBER then we should probably bail out, rather than pretending we succeeded doing a merge build

Reply:
we'll bail out upwards (we check if CIRCLE_PR_NUMBER is defined)

Reply:
Then why do we have an if guard?

Reply:
oooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooooh

Reply:
Comment maybe for us slow people at the back? 0:-)